aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -30,10 +30,10 @@ void THTensor_(zero)(THTensor *r_)
 void THTensor_(maskedFill)(THTensor *tensor, THByteTensor *mask, scalar_t value)
 {
 #ifdef _OPENMP
-  if (!omp_in_parallel()) {
-    int64_t tensor_size = THTensor_(nElement)(tensor);
-    int tensor_contig = THTensor_(isContiguous)(tensor);
-    int mask_contig = THTensor_(isContiguous)(mask);
+  int64_t tensor_size = THTensor_(nElement)(tensor);
+  int tensor_contig = THTensor_(isContiguous)(tensor);
+  int mask_contig = THTensor_(isContiguous)(mask);
+  if (!omp_in_parallel() && tensor_contig && mask_contig) {
```

Comment:
Does `TH_TENSOR_APPLY2_OMP` work with non-contiguous tensors? We use it in other code paths for non-contiguous tensors. Is there something about this operation that makes it different?